core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/PurchasesDelegate.ios.kt

```diff
@@ -45,3 +46,20 @@ private class PurchasesDelegateWrapper(val wrapped: PurchasesDelegate) :
     }
 
 }
+
+/**
+ * On the iOS platform side, the backing field of `Purchases.delegate`, `Purchases.privateDelegate`,
+ * is a `weak var`. On the multiplatform side, we wrap the actual delegate in
+ * `PurchasesDelegateWrapper` to conform to `RCPurchasesDelegateProtocol`. Since that is a private
+ * implementation detail, the only reference held to our wrapper is the `weak var`. This means that
+ * it gets deallocated the first chance it gets. For this reason, we keep a strong reference to our
+ * `PurchasesDelegateWrapper` here. This matches the Android behavior, which keeps a strong
+ * reference on the platform side already.
+ *
+ * **Note**: we cannot make our `PurchasesDelegateWrapper` an `object` directly, as objects cannot
+ * extend `NSObject`. See also
+ * [KT67930](https://youtrack.jetbrains.com/issue/KT-67930/Getting-Crash-while-building-KMM-project-with-XCode-15.3#focus=Comments-27-9796215.0-0)
+ */
+private object PurchasesDelegateStrongReference {
```

Comment:
Hmm it would be great if we could at least tie it to the `Purchases` object lifecycle... So if the `Purchases` object is cleared, this is also cleared... Wdyt?